test/unit/connection_string.test.ts

```diff
@@ -89,6 +94,39 @@ describe('Connection String', function () {
     expect(options.credentials.source).to.equal('$external');
   });
 
+  it('should retain user-specified authSource with no credentials for non-srv URI', function () {
```

Comment:
This test is failing on evg as it expects an empty username but the property does not exist.